plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -110,7 +110,24 @@ export class PersonState {
 
     async update(): Promise<[Person, Promise<void>]> {
         if (!this.processPerson) {
-            const existingPerson = await this.db.fetchPerson(this.teamId, this.distinctId, { useReadReplica: true })
+            let existingPerson = await this.db.fetchPerson(this.teamId, this.distinctId, { useReadReplica: true })
+
+            if (!existingPerson) {
+                // See the comment in `mergeDistinctIds`. We are inserting a row into `posthog_personlessdistinctid`
+                // to note that this Distinct ID has been used in "personless" mode. This is necessary
+                // so that later, during a merge, we can decide whether we need to write out an override
+                // or not.
+                // TODO(perf): Add a cache that tells us we can skip doing the insert
+                const personIsMerged = await this.db.addPersonlessDistinctId(this.teamId, this.distinctId)
+                if (personIsMerged) {
+                    // If `personIsMerged` comes back `true`, it means the `posthog_personlessdistinctid`
+                    // has been updated by a merge (either since we called `fetchPerson` above, plus
+                    // replication lag). We need to check `fetchPerson` again (this time using the leader)
+                    // so that we properly associate this event with the Person we got merged into.
+                    existingPerson = await this.db.fetchPerson(this.teamId, this.distinctId, { useReadReplica: false })
```

Comment:
So what happens if this doesn't return a Person? As the code is written, things should work (it will be treated as a personless event with no actual Person).

Imagine this:

1. `brett` is used as Personless, and added to `posthog_personlessdistinctid`
2. `brett` is merged, and so `posthog_personlessdistinctid` now has `is_merged=true`
3. `brett` (the merged Person) is deleted
4. `brett` event is sent, and lands in this branch because of the row in `posthog_personlessdistinctid` but doesn't find a Person

I need to brush up on exactly how/when we delete Persons. Maybe we can just properly clean up `posthog_personlessdistinctid` rows and call it a day? I think this is *safe* as-is, but it does mean we do needless lookups.